### DISCUSSION TITLE
Problem with userfunc scripts

### DISCUSSION THREAD

**Discussion by jglavrsen** [PiSupply/PiJuice#764] [General]
Help with user scripts
I'm having some trouble running python scripts at events
I have added
/home/pi/script.py
python3 /home/pi/script.py

I have also tested with bash

**Comment by tvoverbeek** [comment]
Need more details.
What did you specify for the event action: USER_FUNCx
and how did you define USER_FUNCn?
Read the Notes in https://github.com/PiSupply/PiJuice/tree/master/Software#user-scripts-menu

**Comment by jglavrsen** [comment]
![image](https://user-images.githubusercontent.com/41836080/126497314-433a2d70-4da1-4bba-865a-306bf750ffbc.png)
![image](https://user-images.githubusercontent.com/41836080/126497398-be0e7437-4837-4a0b-8fab-3517a39c0377.png)

**Comment by tvoverbeek** [comment]
Only `/home/pi/script.py` (i.e. USER_FUNC_3`) works.
The script should be executable and start with `#!/usr/bin/python3`.
Again, read the Notes in https://github.com/PiSupply/PiJuice/tree/master/Software#user-scripts-menu

**Reply by jglavrsen** [reply]
The user 'pi' (the only user there is) is member of pijuice
pi@uit-jgl-oobm04:~ $ getent group pijuice
pijuice:x:1001:pi

The script (user_func1.py - and yes, the scrip is executable)

#!/usr/bin/python3

import sys
import logging

logger = logging.getLogger('user_func1')
hdlr = logging.FileHandler('/home/pi/user_func1.log')
formatter = logging.Formatter('%(asctime)s %(levelname)s %(message)s')
hdlr.setFormatter(formatter)
logger.addHandler(hdlr)
logger.setLevel(logging.INFO)

logger.info(str(sys.argv))
--- End of script ---

**Comment by jglavrsen** [comment]
I found this, and this works
https://learn.pi-supply.com/make/how-to-run-a-user-script-with-the-pijuice-power-events/
